src/main/java/org/jenkinsci/plugins/gwt/jobfinder/JobFinderFactory.java

```diff
@@ -0,0 +1,17 @@
+package org.jenkinsci.plugins.gwt.jobfinder;
+
+import java.util.List;
+import org.jenkinsci.plugins.gwt.FoundJob;
+import org.jenkinsci.plugins.gwt.global.JobFinderConfig;
+
+public final class JobFinderFactory {
```

Comment:
I think you can keep just the one JobFinder and change it a little.

Get the variables `Map<String,String>` with `VariablesResolver`.

And pass the job name list from `JobFinderConfig.get().getFullNameList` to `JobFinderImpersonater`. And let `JobFinderImpersonater` get jobs by name if that list is not empty, else if empty just get all jobs.

Reply:
I think there are two job finder.
1.The old job finder is filter by token, And the precise job finder is filter by the content of request. They cannot exist at the same time, because the requirement. I can't distinguish between the precise job finder can't find the corresponding job and the default job finder, So ,There use the factory to choice the job finder.
2.Convenient for subsequent expansion of job finder，There have the other job finder In the future.